packages/plugins/custom-fields/admin/src/components/ColorPicker/ColorPickerInput/index.js

```diff
@@ -0,0 +1,196 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Stack } from '@strapi/design-system/Stack';
+import { Typography } from '@strapi/design-system/Typography';
+import { Flex } from '@strapi/design-system/Flex';
+import { Box } from '@strapi/design-system/Box';
+import { BaseButton } from '@strapi/design-system/BaseButton';
+import { FocusTrap } from '@strapi/design-system/FocusTrap';
+import { Field, FieldHint, FieldError, FieldLabel, FieldInput } from '@strapi/design-system/Field';
+import { Popover } from '@strapi/design-system/Popover';
+import CarretDown from '@strapi/icons/CarretDown';
+import { useIntl } from 'react-intl';
+import { HexColorPicker } from 'react-colorful';
+
+import getTrad from '../../../utils/getTrad';
+
+const ColorPreview = styled.div`
+  border-radius: 50%;
+  width: 20px;
+  height: 20px;
+  margin-right: 10px;
+  background-color: ${(props) => props.color};
+  border: 1px solid rgba(0, 0, 0, 0.1);
+`;
+
+const ColorPicker = styled(HexColorPicker)`
+  && {
+    width: 100%;
+    height: 152px;
+  }
+
+  .react-colorful__pointer {
+    width: ${({ theme }) => theme.spaces[3]};
+    height: ${({ theme }) => theme.spaces[3]};
+  }
+
+  .react-colorful__saturation {
+    border-radius: ${({ theme }) => theme.spaces[1]};
+  }
+
+  .react-colorful__hue {
+    border-radius: 10px;
+    height: ${({ theme }) => theme.spaces[3]};
+    margin-top: ${({ theme }) => theme.spaces[2]};
+  }
+`;
+
+const ColorPickerToggle = styled(BaseButton)`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+
+  svg {
+    width: ${({ theme }) => theme.spaces[2]};
+    height: ${({ theme }) => theme.spaces[2]};
+  }
+
+  svg > path {
+    fill: ${({ theme }) => theme.colors.neutral500};
+    justify-self: flex-end;
+  }
+`;
+
+const ColorPickerPopover = styled(Popover)`
+  padding: ${({ theme }) => theme.spaces[1]};
+`;
+
+const ColorPickerInput = ({
+  attribute,
+  description,
+  disabled,
+  error,
+  intlLabel,
+  labelAction,
+  name,
+  onChange,
+  required,
+  value,
+}) => {
+  const [showColorPicker, setShowColorPicker] = useState(false);
+  const colorPickerButtonRef = useRef();
+  const { formatMessage } = useIntl();
+  const color = value || '#ffffff';
+  const styleUppercase = { textTransform: 'uppercase' };
+
+  const handleBlur = (e) => {
+    e.preventDefault();
+
+    if (!e.currentTarget.contains(e.relatedTarget)) {
+      setShowColorPicker(false);
+    }
+  };
+
+  return (
+    <Field
+      name={name}
+      id={name}
+      // GenericInput calls formatMessage and returns a string for the error
+      error={error}
+      hint={description && formatMessage(description)}
+    >
+      <Stack spacing={1}>
+        <Flex>
```

Comment:
You'll want something like

```js
{labelAction && <Box paddingLeft={1}>{labelAction}</Box>}
```

to that Flex, so that i18n can inject the localized icons

Reply:
I believe it is handled by passing `labelAction` to the `action` prop on `FieldLabel`, [Suggestion from Gustav](https://github.com/strapi/strapi/pull/14212#discussion_r954913069)

Reply:
So I think can remove this Flex also now that I think about

Reply:
awesome